vtm.js

```diff
@@ -192,67 +193,110 @@ const app = Vue.createApp({
+    } 
+  },
   template: `
-    <stat-section 
-      :stats="attr">
-    </stat-section>
-    <stat-section 
-      :stats="skills">
-    </stat-section>`
+  <stat-section 
+    :stats="this.attributes"
+    @statsectionchange="this.updateStat($event)"
+    >
+  </stat-section>
+  <stat-section 
+    :stats="this.skills"
+    @statsectionchange="this.updateStat($event)"
+    >
+  </stat-section>`
 });
 
+/** Displays entire section of attributes or skills
+ * receives events from child component and check if change is possible in resources
+ * if changes are possible, emit event to top component to make changes, otherwise don't
+ */
 app.component('stat-section', {
   props: ['stats'],
-  data() {
-    return {
-      resourceCount: null,
-    };
-  },
-/**
-   * Counts values in stats.data, same format as stats.resource
-   *
-   * @param {Number} stats.resource Amount of dots you can allocate
-   * @param {JSON} stats.data Current values
-   * @param {Array} of numbers resourceCount - restricting how many stats can have array index values, for example:
-   * [ 0, -8, 4, 3, 1, 0 ]
-   * first zero - we can have 0 attributes with value 0
-   * -8 - we can have only 1 attribute with value 1 and there are 9 attributes with value 1 currently
-   * 4 - we can have 4 attributes with value 2 and there are zero allocated
-   * 3 - we can have 3 attrs with value 3
-   * ...
-   */
-  created() {
-    let tmp = Array(this.stats.resource.length).fill(0);
-    for (var i = 0; i < this.stats.data.length; i++) {
-      for (var j = 0; j < this.stats.data[i].list.length; j++) {
-        tmp[this.stats.data[i].list[j].value]++;
+  computed: {
+    /**
+     * @returns {Array} of numbers describing how many points are currently assigned
+     * for example [0,9,0,0,0,0] means there are 9 attributes with value 1
+     * while [0,1,4,3,1,0] means 1 attribute with value 1, 4 attributes with value 2,
+     * 3 attributes with value 3, 1 attribute with value 4 and 0 attributes with value 5
+     */
+    allocatedResources() {
+      let tmp = Array(this.stats.resource.length).fill(0);
+      for (var i = 0; i < this.stats.data.length; i++) {
+        for (var j = 0; j < this.stats.data[i].list.length; j++) {
+          tmp[this.stats.data[i].list[j].value]++;
+        }
       }
+      return tmp;
     }
-    tmp.forEach(
-      (element, index) => (tmp[index] = this.stats.resource[index] - element)
-    );
-    this.resourceCount = tmp;
   },
-  template: `    
+  emits: {statsectionchange: null},
+  template: `
     <div class="statSection">
       <h2>{{stats.id}}</h2>
-      <div class="resourceCount">{{resourceCount}}</div>
+      <div class="resourceCount">{{allocatedResources}}</div>
       <stat-category 
         v-for="list in stats.data"
         :key="list.id"
         :categ="list"
-        :resource="this.resourceCount"
-        :scale="stats.resource.length - 1">
+        :scale="stats.resource.length - 1"
+        @statcategorychange="          
```

Comment:
toto by mohlo byt prehladnejsie ako methoda

Reply:
dobry napad, pozrem ako to urobit